packages/webcomponentsjs/ts_src/platform/promise.ts

```diff
@@ -0,0 +1,63 @@
+/**
+@license
+Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at
+http://polymer.github.io/LICENSE.txt The complete set of authors may be found at
+http://polymer.github.io/AUTHORS.txt The complete set of contributors may be
+found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by Google as
+part of the polymer project is also subject to an additional IP rights grant
+found at http://polymer.github.io/PATENTS.txt
+*/
+
+// No easy way to provide typings for a file imported this way from
+// node modules. So we'll just ignore the import and cast locally.
+// @ts-ignore
+import PromisePolyfillRaw from '../../node_modules/promise-polyfill/src/index.js';
```

Comment:
The other file removed the `../../node_modules/` part, would that work here too? Also, does TS write out modified import specifiers itself or leave them in a Node-like format?

Reply:
Looks like we can remove that part, yeah. I didn't mean to commit the version with it removed from the other file, but it looks like closure compiler can handle this style.

> Also, does TS write out modified import specifiers itself or leave them in a Node-like format?

TS will never modify import specifiers, it always passes them through untouched.

Reply:
> TS will never modify import specifiers, it always passes them through untouched.

I was going to say "in that case, we probably shouldn't since this will change these packages from being web-compatible to requiring a build step" but it looks like the file in promise-polyfill that we link to wasn't web-compatible in the first place because the specifiers it uses internally assume node-style lookups.

Reply:
Yeah.

There's also a perspective where all of this code is specifically for the oldest browsers, which don't have native modules anyways.